src/Microsoft.DotNet.Build.Tasks/GatherFoldersToRestore.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Text;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GatherDirectoriesToRestore : Task
+    {
+        [Required]
+        public string[] RootDirectories { get; set; }
+
+        [Output]
+        public string[] DirectoriesToRestore { get; set; }
+
+        public int DirectoryListLength = 7000;
```

Comment:
I meant to make this configurable, but forgot to make it a property.  *NIX OS's appear to have no fixed limit on the command line.  Note today with a limit of 7000 we only generate one command line invocation.